action.yaml

```diff
@@ -72,17 +82,43 @@ runs:
           PRE_COMMIT_ARGS="--files ${{ steps.changed-files.outputs.all_changed_files }}"
           echo "PRE_COMMIT_ARGS=$PRE_COMMIT_ARGS" >> "$GITHUB_ENV"
         fi
+
+    - name: Compute pre-commit cache key
+      if: inputs.s3-bucket-name != ''
+      id: pre_commit_cache_key
+      run: |
+        echo "key=${{ hashFiles('**/.pre-commit-config.yaml') }}" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Load cached pre-commit hooks if available
+      if: inputs.s3-bucket-name != ''
+      uses: everpcpc/actions-cache@v2
+      id: read_cache
+      with:
+        bucket: ${{ inputs.s3-bucket-name }}
+        use-fallback: false
+        path: ${{ runner.temp }}/${{ env.cache-name }}-${{ steps.pre_commit_cache_key.outputs.key }}
+        key: ${{ env.cache-name }}-${{ steps.pre_commit_cache_key.outputs.key }}
+        restore-keys: ${{ env.cache-name }}-
+      env:
+        AWS_REGION: ${{ inputs.s3-bucket-region }}
+        cache-name: ${{ github.event.repository.name }}/cache-pre-commit
+
     - name: Pre-commit (action)
       # Same as above, this will install and run pre-commit for us
       if: env.PRE_COMMIT_BIN == null && steps.pre-commit-config.outputs.exists != 'false'
       uses: pre-commit/action@v3.0.1
       with:
         extra_args: ${{ env.PRE_COMMIT_ARGS }}
+      env:
+        PRE_COMMIT_HOME: ${{ runner.temp }}/${{ github.event.repository.name }}/cache-pre-commit-${{ steps.pre_commit_cache_key.outputs.key }}
     - name: Pre-commit (from PATH)
       # Run pre-commit directly if we found it on the PATH
       if: env.PRE_COMMIT_BIN != null && steps.pre-commit-config.outputs.exists != 'false'
       shell: bash
       run: pre-commit run --show-diff-on-failure --color=always ${{ env.PRE_COMMIT_ARGS }}
+      env:
+        PRE_COMMIT_HOME: ${{ runner.temp }}/${{ github.event.repository.name }}/cache-pre-commit-${{ steps.pre_commit_cache_key.outputs.key }}
+
     - name: Restore commitlint config
```

Comment:
I wanted to add a step that deletes the pre-commit cache folder from the tmp folder, but since composite actions can't have a post step, the deletion might run before the actions-cache process to upload to s3 the files so not adding it